app/src/googlevr/java/org/mozilla/vrbrowser/PlatformActivity.java

```diff
@@ -170,8 +177,27 @@ void setImmersiveSticky() {
                                 | View.SYSTEM_UI_FLAG_IMMERSIVE_STICKY);
     }
 
+
     void queueRunnable(Runnable aRunnable) {
-        mView.queueEvent(aRunnable);
+        if (mSurfaceCreated) {
+            mView.queueEvent(aRunnable);
+        } else {
+            synchronized (mPendingEvents) {
```

Comment:
Are events getting queued off of the main UI thread?

Reply:
`queueRunnable` is always called from the main thread. But the pendingEvents array is iterated also when the surfaceCreated callback is received to notify the pending events